demosplan/DemosPlanCoreBundle/ResourceTypes/ProcedureMapSettingResourceType.php

```diff
@@ -0,0 +1,291 @@
+                $masterTemplateMapSetting = $this->masterTemplateService->getMasterTemplate()->getSettings();
+
+                return $this->convertFlatListToCoordinates($masterTemplateMapSetting->getMapExtent(), 4);
+            });
+
+        $configBuilder->useGlobaInformationUrl
+            ->readable(false, fn (ProcedureSettings $procedureSetting): bool => $this->globalConfig->isMapGetFeatureInfoUrlGlobal());
+
+        return $configBuilder;
+    }
+
+    protected function getSetting(string $settingName, ProcedureSettings $procedureSetting): ?Setting
+    {
+        $settings = $this->contentService->getSettings(
+            $settingName,
+            SettingsFilter::whereProcedureId($procedureSetting->getProcedure()->getId())->lock(),
+            false);
+        Assert::countBetween($settings, 0, 1);
+
+        return array_pop($settings);
+    }
+
+    protected function convertStartEndCoordinatesToFlatList(?array $coordinates): string
+    {
+        return null === $coordinates ? '' : implode(',', [
+            $coordinates['start']['latitude'],
+            $coordinates['start']['longitude'],
+            $coordinates['end']['latitude'],
+            $coordinates['end']['longitude']]);
+    }
+
+    protected function convertCoordinatesToFlatList(?array $coordinates): string
+    {
+        return null === $coordinates ? '' : implode(',', [
+            $coordinates['latitude'],
+            $coordinates['longitude']]);
+    }
+
+    protected function convertFlatListToCoordinates(string $rawCoordinateValues, $expectedCoordinatePair): ?array
+    {
+        if ('' === $rawCoordinateValues) {
+            return null;
+        }
+
+        $rawCoordinateValues = explode(',', $rawCoordinateValues);
+        $coordinateValues = [];
+
+        foreach ($rawCoordinateValues as $value) {
+            $coordinateValues[] = (float) $value;
+        }
+
+        Assert::count($coordinateValues, $expectedCoordinatePair);
+
+        if (2 === $expectedCoordinatePair) {
+            return [
+                'latitude'  => $coordinateValues[0],
+                'longitude' => $coordinateValues[1],
+            ];
+        } elseif (4 === $expectedCoordinatePair) {
+            return [
+                'start' => [
+                    'latitude'  => $coordinateValues[0],
+                    'longitude' => $coordinateValues[1],
+                ],
+                'end' => [
+                    'latitude'  => $coordinateValues[2],
+                    'longitude' => $coordinateValues[3],
+                ],
+            ];
+        } else {
+            throw new InvalidArgumentException('Expected exactly two or four coordinate values');
+        }
+    }
+
+    protected function convertCoordinatesToJson(?array $coordinates): ?string
+    {
+        return null === $coordinates ? null : json_encode($coordinates, JSON_THROW_ON_ERROR);
+    }
+
+    protected function convertJsonToCoordinates(string $rawCoordinateValues): ?array
```

Comment:
`$rawCoordinateValues` may be nullable?

Reply:
@dresslerdemos it is emtpy string not nullable, and that is already checked on the method

Reply:
In this case something seems wrong with `$procedureSettings->setTerritory($this->convertCoordinatesToJson($territory));`, because `convertCoordinatesToJson` may return `null`, but `setTerritory` must not be provided with `null`, as its set property `ProcedureSettings::territory` is not a nullable column.

Reply:
d9cdd5945b1d5f6b6e8b9e96720e102950e6e9c4